controller/src/main/java/io/pravega/controller/fault/ControllerClusterListener.java

```diff
@@ -42,23 +43,27 @@
     private final ExecutorService executor;
     private final Optional<ControllerEventProcessors> eventProcessorsOpt;
     private final TaskSweeper taskSweeper;
+    private final Optional<TxnSweeper> txnSweeperOpt;
 
     public ControllerClusterListener(final Host host, final Cluster cluster,
                                      final Optional<ControllerEventProcessors> eventProcessorsOpt,
                                      final TaskSweeper taskSweeper,
+                                     final Optional<TxnSweeper> txnSweeperOpt,
                                      final ExecutorService executor) {
         Preconditions.checkNotNull(host, "host");
         Preconditions.checkNotNull(cluster, "cluster");
         Preconditions.checkNotNull(executor, "executor");
         Preconditions.checkNotNull(eventProcessorsOpt, "eventProcessorsOpt");
         Preconditions.checkNotNull(taskSweeper, "taskSweeper");
+        Preconditions.checkNotNull(txnSweeperOpt, "txnSweeperOpt");
 
         this.objectId = "ControllerClusterListener";
         this.host = host;
         this.cluster = cluster;
         this.executor = executor;
         this.eventProcessorsOpt = eventProcessorsOpt;
         this.taskSweeper = taskSweeper;
+        this.txnSweeperOpt = txnSweeperOpt;
     }
 
     @Override
```

Comment:
Agree, Cluster interface is problematic, as it does not return specific exceptions. Filed an issue #1433 to track this problem.